instrumentation-docs/src/main/java/io/opentelemetry/instrumentation/docs/DocGeneratorApplication.java

```diff
@@ -26,25 +26,37 @@ public class DocGeneratorApplication {
 
   private static final Logger logger = Logger.getLogger(DocGeneratorApplication.class.getName());
 
-  public static void main(String[] args) throws JsonProcessingException {
-    FileManager fileManager = new FileManager("instrumentation/");
+  public static void main(String[] args) throws IOException {
+    // Identify path to repo so we can use absolute paths
+    String baseRepoPath = getRepoPath();
+
+    FileManager fileManager = new FileManager(baseRepoPath);
     List<InstrumentationModule> modules = new InstrumentationAnalyzer(fileManager).analyze();
 
     try (BufferedWriter writer =
         Files.newBufferedWriter(
-            Paths.get("docs/instrumentation-list.yaml"), Charset.defaultCharset())) {
+            Paths.get(baseRepoPath + "/docs/instrumentation-list.yaml"),
+            Charset.defaultCharset())) {
       writer.write("# This file is generated and should not be manually edited.\n");
       writer.write("# The structure and contents are a work in progress and subject to change.\n");
       writer.write(
           "# For more information see: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/13468\n\n");
       YamlHelper.generateInstrumentationYaml(modules, writer);
-    } catch (IOException e) {
-      logger.severe("Error writing instrumentation list: " + e.getMessage());
     }
 
     printStats(modules);
   }
 
+  private static String getRepoPath() throws IOException {
+    URL resource = DocGeneratorApplication.class.getClassLoader().getResource("");
```

Comment:
wdyt about passing the root directory from gradle via system property like in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4107e8a23aead80b5996b871964485dcfcd3c9de/test-report/build.gradle.kts#L22 If the system property is not set you could assume that main class was run from intellij and there the current dir is project root.

Reply:
i think that is a lot cleaner, thank you for the suggestion

Reply:
The weird thing about `getClassLoader().getResource("")` is that afaik not all of the class loaders return anything for it. Here it would be ok, but in a real library it could cause surprises.